test/integration/puppetlabs/general_puppet/general_puppet_int_test.clj

```diff
@@ -154,3 +154,18 @@
         (is (= 500 (:status response)))
         (is (re-matches #".*Cannot retrieve code content because the \"versioned-code\.code-content-command\" setting is not present in configuration.*"
                         (:body response))))))))
+
+(deftest ^:integration static-file-content-works-with-legacy-auth
+  (testing "The static_file_content endpoint works even if legacy-auth is enabled."
+    ;; with-test-logging is used here to supress a warning about running with legacy auth enabled.
+    (logging/with-test-logging
+     (bootstrap/with-puppetserver-running
+      app
+      {:jruby-puppet {:use-legacy-auth-conf true}
+       :versioned-code
+       {:code-content-command (script-path "echo")
+        :code-id-command (script-path "echo")}}
+      (testing "for legacy puppet routes"
+        (let [response (get-static-file-content "modules/foo/files/bar?code_id=foobar&environment=test")]
```

Comment:
Do we have a test already that would cover ensuring that the static_file_content endpoint works if legacy-auth is disabled?

Also, is there a way we could write this test where we could ensure that tk-auth is evaluating the route in both cases - as opposed to just automatically always granting access to the endpoint?  Maybe you could do this by bootstrapping Puppet Server with a custom tk-auth rule where you could validate that a 200 (success) corresponds to a case where the auth rule would allow the request vs. a 403 (failure) would correspond to a case where the auth rule would deny the request?